Shared/Supporting Files/Views/FavoritesView.swift

```diff
@@ -0,0 +1,153 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+
+struct FavoritesView: View {
+    /// All samples retrieved from the Samples directory.
+    let samples: [Sample]
+    
+    /// A Boolean value indicating whether the add favorite sheet is showing.
+    @State private var isShowingSheet = false
+    
+    /// The names of the favorited samples loaded from user defaults.
+    @AppStorage(UserDefaults.favoritedSamplesKey) private var favoritedNames: [String] = []
+    
+    /// A list of the favorited samples.
+    private var favoritedSamples: [Sample] {
+        favoritedNames.compactMap { name in
+            samples.first(where: { $0.name == name })
+        }
+    }
+    
+    var body: some View {
+        List {
+            ForEach(favoritedSamples, id: \.name) { sample in
+                SampleLink(sample)
+            }
+            .onMove { fromOffsets, toOffset in
+                favoritedNames.move(fromOffsets: fromOffsets, toOffset: toOffset)
+            }
+            .onDelete { atOffsets in
+                favoritedNames.remove(atOffsets: atOffsets)
+            }
+        }
+        .listStyle(.sidebar)
+        .toolbar {
+            ToolbarItemGroup(placement: .navigationBarTrailing) {
+                EditButton()
+                
+                Button {
+                    isShowingSheet = true
+                } label: {
+                    Image(systemName: "plus")
+                }
+                .sheet(isPresented: $isShowingSheet) {
+                    AddFavoriteView(samples: samples)
+                }
+            }
+        }
+    }
+}
+
+private extension FavoritesView {
+    /// A view to add a favorite sample from a searchable list.
+    struct AddFavoriteView: View {
+        /// All samples retrieved from the Samples directory.
+        let samples: [Sample]
```

Comment:
I think the current list is OK. It would be better to filter out already favorited samples from the list though.

Reply:
The full list is passed here because it is needed for the `AddFavoriteView`. Also, it makes so `CategoriesView` doesn't have handle an `AppStorage` variable, which is a bit unrelated to showing the categories. (This will be updated as part of `swift/issues/4842`, after this is merged.)